File: C++/31-整数中1出现的次数.cpp
class Solution
{
public:
    int NumberOf1Between1AndN_Solution(int n)
    {
        int res = 0, m = 1;
        while (m <= n)
        {
            int a = n / m, b = n % m;
            res += (a + 8) / 10 * m + (a % 10 == 1) * (b + 1);
            m *= 10;
        }
        return res;
    }
};

File: Python/31-整数中1出现的次数.py
# -*- coding:utf-8 -*-
class Solution:
    def NumberOf1Between1AndN_Solution(self, n):
        # write code here
        res, m = 0, 1
        while m <= n:
            a, b = n // m, n % m
            res += (a + 8) / 10 * m + (a % 10 == 1) * (b + 1)
            m *= 10
        return res